WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterListItem.kt

```diff
@@ -30,35 +118,46 @@ sealed class SubfilterListItem(val type: ItemType) {
         }
     }
 
-    enum class ItemType {
-        SECTION_TITLE,
-        SITE_ALL,
-        SITE,
-        DIVIDER,
-        TAG
+    enum class ItemType constructor(val value: Int) {
+        SECTION_TITLE(0),
+        SITE_ALL(1),
+        SITE(2),
+        DIVIDER(3),
+        TAG(4);
+
+        companion object {
+            fun fromInt(value: Int): ItemType? = values().firstOrNull { it.value == value }
+        }
     }
 
     data class SectionTitle(override val label: UiString) : SubfilterListItem(SECTION_TITLE)
 
     object Divider : SubfilterListItem(DIVIDER)
 
     data class SiteAll(
-        override val label: UiString,
         override var isSelected: Boolean = false,
-        override val onClickAction: (filter: SubfilterListItem) -> Unit
-    ) : SubfilterListItem(SITE_ALL)
+        override var onClickAction: ((filter: SubfilterListItem) -> Unit)?
+    ) : SubfilterListItem(SITE_ALL) {
+        override val label: UiString = UiStringRes(R.string.reader_filter_all_sites)
+    }
 
     data class Site(
-        override val label: UiString,
         override var isSelected: Boolean = false,
-        override val onClickAction: (filter: SubfilterListItem) -> Unit,
+        override var onClickAction: ((filter: SubfilterListItem) -> Unit)?,
```

Comment:
I think it might really be a better solution to have a different object that is stored to AppPrefs which you'd use to build the `SubfilterListItem`. The mapper could be called directly in the `ViewModel` and this way you wouldn't need to make this a `var`. What do you think?

Reply:
Makes sense 👍! Introduced a `SubfilterListItemMapper`